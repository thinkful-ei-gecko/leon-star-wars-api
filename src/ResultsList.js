import React from 'react'

export default function ResultsList(props) {

  function mapResults() {
    let results = props.results || [];
    let count = 0;

    if (!results[0]) {
      return <p id="noResults">No results!</p>
    } else {
      return results.map(result => {
          count++;
          return <li key={result}><span className="count">{count}.</span> {result}</li>;
    })
  }
}

  return(
    <ul>
      {mapResults()}
    </ul>
  );
}