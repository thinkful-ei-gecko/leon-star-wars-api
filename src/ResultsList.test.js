import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ResultsList from './ResultsList'

describe('<ResultsList />', () => {
  it('renders empty fine with no results', () => {
    const wrapper = shallow(<ResultsList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders with results', () => {
    const wrapper = shallow(<ResultsList results={['okay','test','there']} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});




