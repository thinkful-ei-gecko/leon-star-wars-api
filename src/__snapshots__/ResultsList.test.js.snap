// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsList /> renders empty fine with no results 1`] = `
<ul>
  <p
    id="noResults"
  >
    No results!
  </p>
</ul>
`;

exports[`<ResultsList /> renders with results 1`] = `
<ul>
  <li
    key="okay"
  >
    <span
      className="count"
    >
      1
      .
    </span>
     
    okay
  </li>
  <li
    key="test"
  >
    <span
      className="count"
    >
      2
      .
    </span>
     
    test
  </li>
  <li
    key="there"
  >
    <span
      className="count"
    >
      3
      .
    </span>
     
    there
  </li>
</ul>
`;
